### PR TITLE
feat: implemented duration panel

### PR DESCRIPTION
**Pull request by Xanngo**
Implemented DurationPanel. Does not support editable inputs, range or timePickerOptions.

### CODE REVIEW

File: src/date-picker.vue [hunk 1]
```diff
@@ -153,6 +157,20 @@ export default {
           year: 'YYYY',
           month: 'YYYY-MM',
           time: 'HH:mm:ss',
+          duration: {
+            // Date to String
+            stringify: date => {
+              const today = new Date();
+              return date
```
Comment (growson) [inline src/date-picker.vue:163]: @Xanngo please add newline

File: src/date-picker.vue [hunk 2]
```diff
@@ -153,6 +157,20 @@ export default {
...
+                ? `${date.getDate() - today.getDate()} days, ${date.getHours()} hours`
+                : '';
```
Comment (growson) [inline src/date-picker.vue:165]: @Xanngo initial repo uses different ternary styling?

File: src/time/duration-panel.vue [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      default: 'mx',
+    },
+  },
+  props: {
+    value: {},
+    defaultValue: {
+      default() {
+        const date = new Date();
+        date.setHours(0, 0, 0, 0);
+        return date;
```
Comment (growson) [inline src/time/duration-panel.vue:52]: @Xanngo newline above pls

**Review by growson — CHANGES_REQUESTED** [review]

File: src/time/list-columns.vue [hunk 4]
```diff
@@ -134,6 +143,18 @@ export default {
...
+      const la = generateOptions(31, this.dayStep, this.dayOptions).map(num => {
+        const today = new Date();
+        today.setMilliseconds(this.date.getMilliseconds());
+        today.setSeconds(this.date.getSeconds());
+        today.setMinutes(this.date.getMinutes());
+        today.setHours(this.date.getHours());
+        const value = today.setDate(today.getDate() + num);
+        return { value, text: padNumber(num) };
+      });
+      return la;
```
Comment (growson) [inline src/time/list-columns.vue]: @Xanngo I think we may return directly result of `generateOptions`

File: src/time/duration-panel.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    },
+  },
+  computed: {
+    innerValue() {
+      return getValidDate(this.value, this.defaultValue);
+    },
+    title() {
+      const titleFormat = this.timeTitleFormat;
+      const date = new Date(this.innerValue);
+      return this.formatDate(date, titleFormat);
```
Comment (growson) [inline src/time/duration-panel.vue]: @Xanngo I think these 3 lines above really could be written as a one line without vars declaring

**Review by growson — CHANGES_REQUESTED** [review]

File: src/time/list-columns.vue (hunk 4 above)
Comment (Xanngo) [reply src/time/list-columns.vue]: @growson 
O_O
That was debugging code! :see_no_evil:

File: src/date-picker.vue (hunk 2 above)
Comment (Xanngo) [reply src/date-picker.vue:165]: yes

File: src/date-picker.vue (hunk 1 above)
Comment (Xanngo) [reply src/date-picker.vue:163]: Following initial repo coding styles, as discussed in slack.

File: src/time/duration-panel.vue (hunk 3 above)
Comment (Xanngo) [reply src/time/duration-panel.vue:52]: Following initial repo coding styles, as discussed in slack.

**Review by growson — APPROVED** [review]

**Pull request merged**